ex4/VendingDP.Lib/ItemDetectors.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace VendingDP.Lib
+{
+    public class ItemDetectors : MenuLoader
+    {
+        public CombinatedProduct DetectProduct(string productName)
+        {
+            productName = productName.Trim();
+            RawPreparedCombination rawPreparedCombination = GetRawPreparedCombination(productName);
+            if (rawPreparedCombination != null)
+            {
+                return MakeCombinatedProduct(rawPreparedCombination);
+            }
+            else
```

Comment:
remove `else`, inside the `if` block you return, so the code won't continue to run